lib/screens/home/pages/relation/relation_page.dart

```diff
@@ -140,43 +140,59 @@ class _RelationPageState extends State<RelationPage> {
-                      actions: [
-                        FlatButton(
-                          child: Text("Delete"),
-                          onPressed: () {
-                            bloc.add(TaskDeleted(state.relation, task.id));
-                            Navigator.of(context).pop();
+              return Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.start,
+                    children: [
+                      GestureDetector(
+                        onTap: () {
+                          if (!task.isDone) {
+                            bloc.add(TaskCompleted(state.relation, task.id));
                             showProgressIndicator(context);
-                          },
-                        )
-                      ],
-                    ),
-                  );
-                },
-                child: Row(
-                  children: [
-                    GestureDetector(
-                      onTap: () {
-                        if (!task.isDone) {
-                          bloc.add(TaskCompleted(state.relation, task.id));
-                          showProgressIndicator(context);
-                        } else
-                          context.toast("Task already achieved.");
-                      },
-                      child: Checkbox(
-                        value: task.isDone,
+                          } else
+                            context.toast("Task already achieved.");
+                        },
+                        child: Checkbox(
+                          value: task.isDone,
+                        ),
                       ),
+                      Text(task.description),
+                    ],
+                  ),
+                  IconButton(
+                    icon: Icon(
+                      Icons.delete,
+                      color: Colors.grey[700],
                     ),
-                    Text(task.description),
-                  ],
-                ),
+                    onPressed: () {
+                      showDialog(
+                        context: context,
+                        builder: (context) => AlertDialog(
+                          title: Text("Delete task"),
+                          content: Text("Are you sure you want to delete the task?"),
+                          actions: [
+                            FlatButton(
+                              onPressed: () {
+                                Navigator.of(context).pop();
+                              },
+                              child: Text("Cancel"),
+                            ),
+                            FlatButton(
+                              child: Text("Delete"),
+                              onPressed: () {
+                                // bloc.add(TaskDeleted(state.relation, task.id));
```

Comment:
Not sure why you have commented this...May be this is the cause that the delete task is not executed and the progress bar spins forever.

Reply:
yep. This should be it

Reply:
@bharat-1809 Can you check this part?

Reply:
Oh sorry :sweat_smile:

Reply:
Its done